lib/builderator/interface/packer.rb

```diff
@@ -63,24 +63,10 @@ def initialize(*_)
             json[:builders] << build_hash
           end
 
-          post_processors = []
-
-          # Post-processors should be considered as a sequence
-          Config.profile.current.packer.post_processor.each do |name, post_processor|
-            post_processor_hash = post_processor.to_hash
-
-            # Single, named step in a sequence
-            if post_processor_hash.empty?
-              post_processors << name
-              next
-            end
-
-            # The post-processor's type should be the same as the name
-            post_processor_hash[:type] = name
-
-            post_processors << post_processor_hash
-          end
-          json['post-processors'].push(post_processors)
+          # post_processors must be a List of Rashes.
+          post_processors = Config.profile.current.packer.post_processors.to_a
+          json['post-processors'] = post_processors.first unless post_processors.empty?
```

Comment:
The `:list` with a single element is handled with `:singular`; for example check out `:run_list`:

`attribute :run_list, :type => :list, :singular => :run_list_item`

Not sure if this works for your case but it might be useful to use the same method.